crates/topos-sequencer-subnet-runtime-proxy/src/subnet_runtime_proxy.rs

```diff
@@ -211,52 +213,64 @@ impl SubnetRuntimeProxy {
                     info!(
                         "Processing certificate received from TCE, cert_id={}",
-                        &cert.id
+                        &certificate.id
                     );
 
                     // Verify certificate signature
                     // Well known subnet id is public key for certificate verification
                     // Public key of secp256k1 is 33 bytes, we are keeping last 32 bytes as subnet id
                     // Add manually first byte 0x02
-                    let public_key = cert.source_subnet_id.to_secp256k1_public_key();
+                    let public_key = certificate.source_subnet_id.to_secp256k1_public_key();
 
                     // Verify signature of the certificate
                     match topos_crypto::signatures::verify(
                         &public_key,
-                        cert.get_payload().as_slice(),
-                        cert.signature.as_slice(),
+                        certificate.get_payload().as_slice(),
+                        certificate.signature.as_slice(),
                     ) {
                         Ok(()) => {
-                            info!("Certificate {} passed verification", cert.id)
+                            info!("Certificate {} passed verification", certificate.id)
                         }
                         Err(e) => {
-                            error!("Failed to verify certificate id {}: {e}", cert.id);
+                            error!("Failed to verify certificate id {}: {e}", certificate.id);
                             return;
                         }
                     }
 
                     // Push the Certificate to the ToposCore contract on the target subnet
+                    let span_push_certificate = info_span!("Subnet push certificate call");
+                    span_push_certificate.set_parent(ctx);
                     match SubnetRuntimeProxy::push_certificate(
                         runtime_proxy_config,
                         subnet_client,
-                        &cert,
+                        &certificate,
                         cert_position,
                     )
+                    .with_context(span_push_certificate.context())
+                    .instrument(span_push_certificate)
                     .await
                     {
                         Ok(tx_hash) => {
                             debug!(
                                 "Successfully pushed the Certificate {} to target subnet with tx hash {}",
-                                &cert.id, &tx_hash
+                                &certificate.id, &tx_hash
                             );
                         }
                         Err(e) => {
                             error!(
                                 "Failed to push the Certificate {} to target subnet: {e}",
-                                &cert.id
+                                &certificate.id
```

Comment:
Did you see if this log is part of the `span_push_certificate` span ? Because you define the instrument but the match itself isn't in the scope I guess

Reply:
I wanted it to be in the span of `Subnet Runtime Proxy` as runtime proxy has failed to perform this task. Does it make sense?